src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const act of actions) {
+    switch (act.type) {
+      case 'clear':
+        Object.keys(state).forEach(key => {
+          delete state[key];
+        });
+        break;
+      case 'addProperties':
+        Object.assign(state, act.extraData);
+        break;
+      case 'removeProperties':
+        act.keysToRemove.forEach(element => {
+          delete state[element];
+        });
+        break;
+      default:
+        return 'wrong action in actions';
```

Comment:
use `throw new Error('error description');` for error handling

Reply:
thanks